services/ui_backend_service/tests/integration_tests/utils.py

```diff
@@ -275,12 +275,29 @@ async def _test_list_resources(cli, db: AsyncPostgresDB, path: str, expected_sta
 
     expected_data[:] = map(_fill_missing_resource_data, expected_data)
     assert len(data) == len(expected_data)
-    for i, d in enumerate(data):
+
+    data_to_be_compared = data
+    expected_data_to_be_compared = expected_data
+    if ignore_data_order:
+        assert isinstance(data, list) and isinstance(expected_data, list)
+
+        # If item contains fields A and B, then sort list first by item[A], then item[B]
+        # Exclude approx_keys, because we don't want approx_sort.
+        def _sort_key(r):
+            if isinstance(r, str):
```

Comment:
The data list can contain `str` items sometimes.  We handle it.  If other types show up in future we will need to add more rules - we won't miss it.